### ISSUE TITLE
Disable animations and transitions when reduced motion accessibility setting is enabled

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#15187]
When the “Reduce Motion” setting is enabled in the Accessibility section of Settings on iOS, map views should perform animations and transitions instantaneously. This would ensure parity with GL JS, which added support for CSS’s `prefers-reduced-motion` media query in mapbox/mapbox-gl-js#8494.

The iOS map SDK can detect this setting by calling `UIAccessibilityIsReduceMotionEnabled()` and can detect when the setting changes by observing for `UIAccessibilityReduceMotionStatusDidChangeNotification`. There is a similar user setting on macOS but no corresponding notification or getter that I can find.

One wrinkle is that some developers may have baked in an assumption that an animations lasts for a certain amount of time before starting a different animation. We should recommend that developers use completion handlers to chain animations rather than hard-coding delays.

/cc @mapbox/maps-ios

**Comment by dieterdreist** [comment]
Animations and transitions can be used as eye candy, but they can also be an integral part of an application. Apple's "Reduce Motion" feature is not intended to remove essential functionality (which animations and transitions sometimes are), it should only keep the screen calmer. Turning the feature on should not prevent you from using an app as intended.
Mapbox, please, if you implement this, make it developer configurable (respect / ignore setting) or time dependent (animations with longer than x seconds duration should not be prevented).

**Comment by 1ec5** [comment]
We should probably match how MapKit respects (or doesn’t respect) this setting with regard to simpler methods like `-[MKMapView setCamera:animated:]`. On the other hand, MapKit doesn’t have any precedent for longer animations; its animations are fixed to a very short length.

Explicitly making the behavior configurable per camera operation would require additional parameters, which would be unfortunate, due to unwieldiness but also due to forcing the developer to make this decision every time they change the camera. Another approach would be to always perform all animations and expect the developer to check `UIAccessibilityIsReduceMotionEnabled()` as needed, but I’m not confident that developers would generally heed that advice.

If the iOS/macOS map SDK could support Core Animation implicit animation (#4352), then we might have more ways to encapsulate reduced motion support without forcing developers to explicitly opt in or out every time.
